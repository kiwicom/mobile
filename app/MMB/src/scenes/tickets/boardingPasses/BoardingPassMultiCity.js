// @flow strict

import * as React from 'react';
import { Translation } from '@kiwicom/mobile-localization';
import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
import { TextIcon, StyleSheet, Color } from '@kiwicom/mobile-shared';
import idx from 'idx';

import type { BoardingPassMultiCity as BookingType } from './__generated__/BoardingPassMultiCity.graphql';
import FlightSegments from './FlightSegments';

type Props = {|
  +data: BookingType,
|};

const BoardingPassMultiCity = (props: Props) => {
  const trips = idx(props.data, _ => _.trips) || [];
  return trips.map((trip, index) => (
    /* TODO: Icons should change color */
    <FlightSegments
      key={index}
      data={trip}
      icon={<TextIcon code="&#xe079;" style={styles.outboundIcon} />}
      iconTitle={
        <Translation
          passThrough={idx(trip, _ => _.arrival.airport.city.name)}
        />
      }
    />
  ));
};

export default createFragmentContainer(
  BoardingPassMultiCity,
  graphql`
    fragment BoardingPassMultiCity on BookingMulticity {
      trips {
        arrival {
          airport {
            city {
              name
            }
          }
        }
        ...FlightSegments
      }
    }
  `,
);

const styles = StyleSheet.create({
  outboundIcon: {
    color: Color.brand,
    transform: [{ rotate: '90deg' }],
  },
});
