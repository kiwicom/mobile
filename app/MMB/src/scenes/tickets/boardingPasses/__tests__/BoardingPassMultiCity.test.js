// @flow strict

import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { BoardingPassMultiCity } from '../BoardingPassMultiCity';

const renderer = new ShallowRenderer();

const props = {
  trips: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
};
describe('BoardingPassMultiCity', () => {
  it('renders even with many trips', () => {
    expect(
      // $FlowExpectedError: Passing just what is needed to test that the component renders
      renderer.render(<BoardingPassMultiCity data={props} />),
    ).toMatchSnapshot();
  });
});
