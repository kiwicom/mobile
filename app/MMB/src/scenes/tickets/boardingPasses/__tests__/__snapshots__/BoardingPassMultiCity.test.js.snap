// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardingPassMultiCity renders even with many trips 1`] = `
Array [
  <Relay(FlightSegments)
    data={1}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#00a991",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={2}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#fbad18",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={3}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#d50c6a",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={4}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#35dc93",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={5}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#ef7209",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={6}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#863ccd",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={7}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#006dc7",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={8}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#4fb52e",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={9}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#30363d",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={10}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#ff80ce",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={11}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#ff2424",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={12}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#ff2424",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={13}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#ff2424",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
  <Relay(FlightSegments)
    data={14}
    icon={
      <TextIcon
        code=""
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            Object {
              "color": "#ff2424",
            },
          ]
        }
      />
    }
    iconTitle={<Translation />}
  />,
]
`;
